Renamed titles, added links and improved wording

aleon1220 opened the pull request:
changed the title for the 3 options on how to set up AWS IAM structure
1-with CF console
2-With AWS CLI
3-with the AWS IAM console
Listed the main components of the AWS IAM structure
In halyard section changed titles to put them in context by using the AWS provider.
I changed the wording to start with actions to perform e.g. Enable the AWS Provider

In Next steps section
I added links to the other AWS compute services: ECS, EKS and lambda

aleon1220 committed:
renamed titles and added links

changed the title for the 3 options on how to set up AWS IAM structure
1-with CF console
2-With AWS CLI
3-with the AWS IAM console
Listed the main components of the AWS IAM structure
In halyard section changed titles to put them in context by using the AWS provider.
I changed the wording to start with actions to perform e.g. Enable the AWS Provider

In Next steps section
I added links to the other AWS compute services: ECS, EKS and lambda

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -11,13 +11,19 @@ sidebar:
-Use the AWS EC2 Provider if you want to manage EC2 Instances via Spinnaker. Refer to the [AWS Cloud Provider Overview](https://spinnaker.io/setup/install/providers/aws/) to understand how AWS IAM must be set up for the AWS EC2 provider to work.
+Use the AWS EC2 Provider if you want to manage EC2 Instances via Spinnaker. Refer to the [AWS Cloud Provider Overview](https://spinnaker.io/setup/install/providers/aws/) to understand how AWS IAM must be set up with the Spinnaker AWS EC2 provider.
+
+Spinnaker will use an [AWS IAM structure](https://aws.amazon.com/iam/) of users, roles, policies, etc to access AWS services and resources securely. There are 3 options to set up the AWS IAM structure
```
```suggestion
Spinnaker will use an [AWS IAM structure](https://aws.amazon.com/iam/) of users, roles, policies, and so on, to access AWS services and resources securely. There are 3 options to set up the AWS IAM structure
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -11,13 +11,19 @@ sidebar:
+
+Spinnaker will use an [AWS IAM structure](https://aws.amazon.com/iam/) of users, roles, policies, etc to access AWS services and resources securely. There are 3 options to set up the AWS IAM structure
+
+1. AWS CloudFormation templates deployed with the CloudFormation Console
```
Maybe hyperlink these to their respective headings below?

dorbin replied:
I meant link each of these three list items to the headings below, in the article (though with headings like these, the auto-generated anchor will be complex. The current anchor for option 1 (on the published site, before these changes) is `#option-1--use-aws-console-to-configure-aws`

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -11,13 +11,19 @@ sidebar:
 account](https://aws.amazon.com/account/){:target="\_blank"}.
 
-## Option-1 : Use AWS Console to configure AWS
+## Option-1 : Configure with AWS CloudFormation Console
```
```suggestion
## Option 1: Configure with AWS CloudFormation Console
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -45,7 +51,7 @@ Use this option to deploy Spinnaker, if you are familar with deployment using [A
 ![](../outputs_cloudformation.png)
 
-## Option-2 : Use AWS CLI to configure AWS
+## Option-2 : Configure with AWS CLI
```
```suggestion
## Option 2: Configure with AWS CLI
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
````diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
 ```
 
-## Option-3 : Use AWS Console UI (Manual Steps) 
+## Option-3 : Configure with AWS IAM Console
````
```suggestion
## Option 3: Configure with AWS IAM Console
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
 
-## Option-3 : Use AWS Console UI (Manual Steps) 
+## Option-3 : Configure with AWS IAM Console
+### Create Roles
```
Are these h3-level sections going to be filled in later?

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
 
 There are 2 options here
-1. Using AWS IAM AccessKey and Secret
+1. Using AWS IAM User with AccessKey and Secret
```
```suggestion
1. Use AWS IAM User with AccessKey and Secret
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
-1. Using AWS IAM AccessKey and Secret
+1. Using AWS IAM User with AccessKey and Secret
 Option number 1 is useful for creation of user with AWS Access Key and secret. This is a common configuration. 
 2. Using AWS IAM Roles
```
```suggestion
2. Use AWS IAM Roles
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
-After the AWS IAM user, roles, policies and trust relationship have been set up, the next step is to add the AWS configurations to Spinnaker via Halyard CLI:
+After the AWS IAM  structure (user, roles, policies and trust relationship) has been set up, the next step is to add the AWS configurations to Spinnaker via Halyard CLI:
 
+The General steps are the following:
```
You can omit this sentence.

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
 3. Enable the AWS provider `hal config provider aws enable`.
 
-### Configure Halyard to use AccessKeys (if configured)
+### Configure Spinnaker AWS provider to use AccessKeys (if using AWS IAM user)
```
```suggestion
### Configure the Spinnaker AWS provider to use AccessKeys (if using AWS IAM user)
```

dorbin replied:
I like these changes a lot, @aleon1220 ! You don't "configure Halyard," you use Halyard to configure Spinnaker!

dorbin commented on setup/install/providers/aws/aws-ec2.md:
````diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
 ```
 
 ## Advanced account settings
+Once you configure the Spinnaker AWS provider you will be able to manage and deploy EC2 resources with Spinnaker.
````
```suggestion
After you configure the Spinnaker AWS provider you can manage and deploy EC2 resources with Spinnaker.
```

aleon1220 replied:
Thanks. More proper English there. Thank you.

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
 
-You can view the available configuration flags for AWS within the
-[Halyard reference](/reference/halyard/commands#hal-config-provider-aws-account-add).
+You can view the available configuration flags for Spinnaker AWS provider within the
```
```suggestion
You can view the available configuration flags for the Spinnaker AWS provider in the
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
-You can view the available configuration flags for AWS within the
-[Halyard reference](/reference/halyard/commands#hal-config-provider-aws-account-add).
+You can view the available configuration flags for Spinnaker AWS provider within the
+[Halyard AWS provider reference](/reference/halyard/commands#hal-config-provider-aws-account-add).
```
```suggestion
the [Halyard command reference](/reference/halyard/commands#hal-config-provider-aws-account-add).
```

dorbin commented on setup/install/providers/aws/aws-ec2.md:
```diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
-provider](/setup/install/providers/), but otherwise you're ready to
-[choose an environment](/setup/install/environment/)
+Optionally, you can enable other AWS Providers:
+[set up Amazon's Elastic Container
```
Andres, can you set these as a bulleted list?

aleon1220 replied:
yes

dorbin reviewed (commented):
Nice! I left a bunch of edits though. :)

aleon1220 committed:
Update setup/install/providers/aws/aws-ec2.md

great. Thanks

Co-authored-by: Dave Dorbin <ddorbin@google.com>

aleon1220 committed:
Update setup/install/providers/aws/aws-ec2.md

Co-authored-by: Dave Dorbin <ddorbin@google.com>

aleon1220 committed:
Update setup/install/providers/aws/aws-ec2.md

agree is best to remove extra character

Co-authored-by: Dave Dorbin <ddorbin@google.com>

aleon1220 committed:
Update setup/install/providers/aws/aws-ec2.md

agree.

Co-authored-by: Dave Dorbin <ddorbin@google.com>

aleon1220 committed:
Update setup/install/providers/aws/aws-ec2.md

yes.

Co-authored-by: Dave Dorbin <ddorbin@google.com>

aleon1220 committed:
Added bulletpoints for Next Steps

Added bullet points
Changed wording to include words "With Spinnaker" great for search engines
changed wording in the last line

aleon1220 reviewed (commented):
updated changes.

aleon1220 committed:
Merge branch 'master' into patch-4

aleon1220 commented:
@dorbin Thanks Dave for your suggestions. I have planned some extra improvements to the complete AWS Provider (EC2, ECS, EKS, Lambda). I have been doing progressively and in small chunks. Thanks.
If you have improvement ideas i am happy to read about them.

aleon1220 committed:
Merge branch 'master' into patch-4

aleon1220 commented:
OK I have created a new PR https://github.com/spinnaker/spinnaker.github.io/pull/1996 with the changes from this PR1964
Hopefully it can be easily merged...
